Source/WebCore/html/canvas/GPUCanvasContextCocoa.mm

```diff
@@ -248,13 +248,17 @@ static bool equalConfigurations(const auto& a, const auto& b)
         && a.colorSpace     == b.colorSpace;
 }
 
-static DestinationColorSpace toWebCoreColorSpace(const GPUPredefinedColorSpace& colorSpace)
+static DestinationColorSpace toWebCoreColorSpace(const GPUPredefinedColorSpace& colorSpace, const GPUCanvasToneMapping& toneMapping)
 {
     switch (colorSpace) {
     case GPUPredefinedColorSpace::SRGB:
-        return DestinationColorSpace::SRGB();
+        return toneMapping.mode == GPUCanvasToneMappingMode::Standard ? DestinationColorSpace::SRGB() : DestinationColorSpace::ExtendedSRGB();
     case GPUPredefinedColorSpace::DisplayP3:
-        return DestinationColorSpace::DisplayP3();
+#if ENABLE(PREDEFINED_COLOR_SPACE_DISPLAY_P3)
+        return toneMapping.mode == GPUCanvasToneMappingMode::Standard ? DestinationColorSpace::DisplayP3() : DestinationColorSpace::ExtendedDisplayP3();
+#else
+        return DestinationColorSpace::SRGB();
```

Comment:
Shouldn't you handle extended in this case as well?

Reply:
yes good idea, it's 1 on Mac and iOS family but I think it is 0 on Mac Catalyst